.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,42 @@
+<!--- Provide a general summary of your changes in the Title above -->
+
+## Description
+<!--- Describe your changes in detail -->
+
+## Checklist:
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
+- [ ] **I am adding a new extension to an existing language.**
+  - [ ] The new extension is used in hundreds of repositories on GitHub.com
+    - Search results for each extension:
+      <!-- Replace FOOBAR with the new extension. Repeat for each extension added. -->
+      - https://github.com/search?utf8=%E2%9C%93&q=extension%3AFOOBAR+NOT+nothack&type=Code&ref=searchresults
+  - [ ] I have included a real-world usage sample for all extensions added in this PR:
+    - Sample source(s):
+      - [URL to each sample source, if applicable]
+    - Sample license(s):
+  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.
```

Comment:
Given that the above points are mandatory, this one feels mandatory too. Do we want to make adding heuristic rules mandatory, or should we discuss this with contributors on a case-by-case basis?

Reply:
I think we should keep it optional as we don't always need a heuristic and suggest it's only optional in CONTRIBUTING.md